protocol/src/lib.rs

```diff
@@ -834,10 +834,17 @@ impl Encoder<Message> for CrucibleEncoder {
         dst: &mut BytesMut,
     ) -> Result<(), Self::Error> {
         let len = CrucibleEncoder::serialized_size(&m)?;
+        if len > MAX_FRM_LEN {
```

Comment:
Nit: can you add a test for this behavior?

Reply:
done in ae64683, thanks!